Automattic/jetpack
# Social: Use core endpoint for dismissing notices

This change updates our notice dismissal logic, so it uses `register_setting` and the core endpoint coming with it.
We have to support the deprecated Jetpack endpoint, so I flagged that with the current versions so we can get rid of it 3 versions later.

## Proposed changes:

* Added the handler class + tests
* Changed the initial states so they use the new getter

### Other information:


## Jetpack product discussion


https://github.com/orgs/Automattic/projects/733/views/2?pane=issue&itemId=46761193
## Does this pull request change what data or activity we track or use?


No.
## Testing instructions:






* Delete the dismissed_notices with `jetpack docker wp option delete jetpack_social_dismissed_notices`
* With auto-conversion being on, create a new post and upload a 10Mb jpg image which should be invalid but convertible. You should see the notice that it will be converted.
* Click `Got it`, then refresh the page and select the image again. You should not see the notice anymore

Remove the advanced plan.
* Go to the editor and dismiss the advancedPlan upgrade nudge
* Make sure both are stored correctly in the option

`jetpack docker wp option get jetpack_social_dismissed_notices`
You can also check it from the console: https://developer.wordpress.com/docs/api/console/
Run a query like this with your site ID:
![CleanShot 2024-01-09 at 11 37 16 png](https://github.com/Automattic/jetpack/assets/36671565/e9a0277f-2196-465b-aad7-0c50468ac4e3)
